### DISCUSSION TITLE
"Too many fields. Maximum number of fields is 1000" - How to change Starlette multipart-form default in FastAPI?

### DISCUSSION THREAD

**Discussion by will-franceys** [fastapi/fastapi#9634] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# requirements.txt
fastapi>=0.92.0
starlette>=0.25.0
python-multipart
```

```python
# app.py

from fastapi import FastAPI, UploadFile

app = FastAPI()

@app.post("/")
def files(files: list[UploadFile]):
    return "OK"
```

```python
# test.py

from starlette.testclient import TestClient

from app import app

client = TestClient(app)

def test_that_passes():
    files = [("files", ("foo.txt", "foo"))] * 1000
    response = client.post("/", files=files)
    assert response.status_code == 200


def test_that_fails():
    # Send more files than default limit
    files = [("files", ("foo.txt", "foo"))] * 1001
    response = client.post("/", files=files)
    # Fails with {'detail': 'Too many files. Maximum number of files is 1000.'}
    assert response.status_code == 200
```

### Description

Summary:
- API fails when loading more than 1000 files in a multipart-form
- Starlette introduced a default of 1000 files in [starlette==0.25.0](https://github.com/encode/starlette/commit/8c74c2c8dba7030154f8af18e016136bea1938fa) 
- 🤔 How can this new default be set through FastAPI?

Steps to reproduce:
- Run the scripts above, note the first test with a 1000 files and the second with 1001 files being sent

What I expected to see:
- Pass for both tests

What I saw instead:
- The second test fails as it goes above the new Starlette limit

### Operating System

Linux

### FastAPI Version

0.96.0

### Python Version

3.10.0

### Additional Context

- Currently pinning to older versions of fastapi and starlette is the workaround
- Would require to pass through `max_fields` [here](https://github.com/tiangolo/fastapi/blob/155fc5e24e4bfde690dd2314c02d93b92ca5d78b/fastapi/routing.py#L195) from `body = await request.form()` to `body = await request.form(max_files=max_files)`

**Comment by will-franceys** [comment]
PR to prompt discussion - https://github.com/tiangolo/fastapi/pull/9640

**Comment by YuriiMotov** [comment]
Let's close this and track it there: https://github.com/fastapi/fastapi/discussions/13557
